addon/components/file-picker.js

```diff
@@ -73,13 +73,39 @@ export default Component.extend({
     }
   },
 
-  handleFiles: function(files) {
+  _invalidExtension: function(files) {
+    let accept = this.get('accept');
+
+    if (accept === '*') {
+      return;
+    }
+
+    let validExtensions = accept.split(',');
+
+    let fileExtensions = files.map(file => `.${file.filename.split('.').slice(-1)[0]}`);
+
+    return fileExtensions.some(extension => validExtensions.indexOf(extension) === -1);
+  },
+
+  _validate: function(files) {
     if (typeof(this.filesAreValid) === 'function') {
       if (!this.filesAreValid(files)) {
         return;
       }
     }
 
+    if (this._invalidExtension(files)) {
```

Comment:
@timmcanty can you add a setting `validateExtension` default can be `true` and add it to the readme`s Options section?